compiler/rustc_span/src/span_encoding.rs

```diff
@@ -61,6 +61,9 @@ use rustc_data_structures::fx::FxIndexSet;
 /// using the callback `SPAN_TRACK` to access the query engine.
 ///
 #[derive(Clone, Copy, Eq, PartialEq, Hash)]
+// FIXME(@lcnr): Enable this attribute once the bootstrap
+// compiler knows of `rustc_pass_by_value`.
+// #[cfg_attr(not(bootstrap), rustc_pass_by_value)]
```

Comment:
```suggestion
#[cfg_attr(not(bootstrap), rustc_pass_by_value)]
```
Why doesn't this work? Isn't `not(bootstrap)` supposed to specifically address this case?

Reply:
`&Span` is used incredibly frequently, so fixing it takes a while. As long as `rustc_pass_by_value` is ignored in the bootstrap compiler, it isn't as nice to deal with that.

It means that the lint will only trigger when compiling the stage 2 compiler, which most people don't tend to do before submitting their PR, causing CI to break. So it would work, but would imo be too annoying to be worth it.

Reply:
Could you say in the comment that not all uses of `Span` are fixed, and it's tedious to fix them until it can be done at stage 0-1 when stage 0-1 supports `#[rustc_pass_by_value]`?
Right now it's entirely unclear from the comment why the attribute is commented out.